File: 125-ValidPalindrome/125-ValidPalindrome.py
# Last updated: 09/10/2025, 09:40:12
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        
        s4 = re.sub(r'[^a-zA-Z0-9]', '', s).lower()
        print s4
        if all(s4[i] == s4[len(s4)-i-1] for i in range (len(s4))) :
            return True
        return False

        

File: 125-ValidPalindrome/125-ValidPalindrome.cpp
// Last updated: 03/03/2026, 20:06:51
class Solution {
public:
    bool isPalindrome(string s)
    {
        int left = 0;
        int right = s.length() - 1;

        while(left < right)
        {
            while(left < right && !isalnum(s[left]))
                left++;

            while(left < right && !isalnum(s[right]))
                right--;

            if(tolower(s[left]) != tolower(s[right]))
                return false;

            left++;
            right--;
        }

        return true;
    }
};
